sdk/python/kfp/client/auth_test.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2023 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from unittest.mock import MagicMock
+from unittest.mock import patch
+
+from absl.testing import parameterized
+from kfp.client import auth
+
+
+class TestAuth(parameterized.TestCase):
+
+    def test_is_ipython_return_false(self):
+        mock = MagicMock()
+        with patch.dict('sys.modules', IPython=mock):
+            mock.get_ipython.return_value = None
+            self.assertFalse(auth.is_ipython())
+
+    def test_is_ipython_return_true(self):
+        mock = MagicMock()
+        with patch.dict('sys.modules', IPython=mock):
+            mock.get_ipython.return_value = 'Something'
+            self.assertTrue(auth.is_ipython())
+
+    def test_is_ipython_should_raise_error(self):
+        mock = MagicMock()
+        with patch.dict('sys.modules', mock):
+            mock.side_effect = ImportError
+            self.assertFalse(auth.is_ipython())
+
+    @patch('builtins.input', lambda *args:
```

Comment:
Nice mocks! I'll be borrowing this pattern for another case where I have to mock `input`. Thanks!